x-pack/plugins/maps/server/maps_telemetry/maps_telemetry.ts

```diff
@@ -270,26 +257,60 @@ export function buildMapsTelemetry({
     },
   };
 }
-async function getMapSavedObjects(savedObjectsClient: SavedObjectsClientContract) {
-  const mapsSavedObjects = await savedObjectsClient.find({ type: MAP_SAVED_OBJECT_TYPE });
-  return _.get(mapsSavedObjects, 'saved_objects', []);
-}
 
-async function getIndexPatternSavedObjects(savedObjectsClient: SavedObjectsClientContract) {
-  const indexPatternSavedObjects = await savedObjectsClient.find({ type: 'index-pattern' });
-  return _.get(indexPatternSavedObjects, 'saved_objects', []);
+export function getLayerLists(mapSavedObjects: MapSavedObject[]): LayerDescriptor[][] {
+  return mapSavedObjects.map((savedMapObject) => {
+    const layerList =
+      savedMapObject.attributes && savedMapObject.attributes.layerListJSON
+        ? JSON.parse(savedMapObject.attributes.layerListJSON)
+        : [];
+    return layerList as LayerDescriptor[];
+  });
 }
 
 export async function getMapsTelemetry(config: MapsConfigType) {
   const savedObjectsClient = getInternalRepository();
-  // @ts-ignore
-  const mapSavedObjects: MapSavedObject[] = await getMapSavedObjects(savedObjectsClient);
-  const indexPatternSavedObjects: IIndexPattern[] = (await getIndexPatternSavedObjects(
-    // @ts-ignore
-    savedObjectsClient
-  )) as IIndexPattern[];
-  const settings: SavedObjectAttribute = {
-    showMapVisualizationTypes: config.showMapVisualizationTypes,
+
+  // Harvest layers lists from saved maps for each page of results
+  let currentPage = 1;
+  let perPage = 2; // Seed value
+  let total = 3; // Seed value
+  let savedObjects = [];
+  let layerLists: LayerDescriptor[][] = [];
+  while (currentPage <= Math.ceil(total / perPage)) {
```

Comment:
What is this loop checking for? Shouldn't the page logic look at the number of items returned and keep going until the number of items returned is less then page size?

Reply:
Sorry for the delayed follow-up, this PR got back-burnered for a while. Agreed this could be simpler. Logic has been revised and the two similar loops have been consolidated under one function.